python/ray/serve/handle.py

```diff
@@ -193,12 +196,14 @@ def _options(
         if self._router is None and _router_cls == DEFAULT.VALUE:
             self._get_or_create_router()
 
-        return self.__class__(
+        new_handle = self.__class__(
             self.deployment_name,
             handle_options=new_handle_options,
             _router=None if _router_cls != DEFAULT.VALUE else self._router,
-            _is_for_http_requests=self._is_for_http_requests,
+            _request_counter=self.request_counter,
         )
+        new_handle._set_request_protocol(self._request_protocol)
```

Comment:
I would suggest still putting the protocol in the `HandleOptions` class (it's not publicly exposed). Then you don't need to have a separate call here.

Reply:
oh makes sense XD, let me fix it real quick